pallets/nft/src/tests.rs

```diff
@@ -3,159 +3,732 @@ use crate::mock::{Event, *};
+				5,
+			),
+			NftError::NoPermission
+		);
+
+		// create Transferable Unburnable class with start/end restrcition
+		assert_ok!(Nft::create_class(
+			Origin::signed(alice_account.clone()),
+			CID::default(),
+			Properties(ClassProperty::Transferable | ClassProperty::Burnable),
+			10.into(),
+			100.into(),
+			ClassType::Simple(100),
+		));
+
+		run_to_block(2);
+		assert_eq!(System::block_number(), 2);
+
+		//mint 5 instance out of time
+		assert_noop!(
+			Nft::mint(
+				Origin::signed(alice_account.clone()),
+				bob_account.clone(),
+				1,
+				CID::default(),
+				5,
+			),
+			NftError::OutOfCampaignPeriod
+		);
+
+		run_to_block(11);
+		assert_eq!(System::block_number(), 11);
+		//mint 5 instance within time
+		assert_ok!(Nft::mint(
+			Origin::signed(alice_account.clone()),
+			bob_account.clone(),
+			1,
+			CID::default(),
+			5,
+		));
+		assert_eq!(
+			events_filter::<crate::Event::<Test>>()[3],
+			Event::Nft(crate::Event::MintedToken(alice_account.clone(), bob_account.clone(), 1, 5))
+		);
+	});
+}
+
+#[test]
+// Test for Claim type Event:
+// ClaimedToken(T::AccountId, ClassIdOf<T>)
+
+// Test for Error:
+// OutOfCampaignPeriod
+// NonTransferable,
+// ClassClaimedListNotFound
+// UserNotInClaimList
+// TokenAlreadyClaimed
+fn test_claimed_token_event() {
+	new_test_ext().execute_with(|| {
+		let (alice_account, bob_account) = initial_account();
+
+		// root is 0x0c67caf461290cd463e535213f996e32736e65a2063783fde5036b71396dfb0c
 		let merkle_root = [
-			0xa8, 0xa5, 0xec, 0x29, 0xa3, 0xdf, 0x3c, 0x5a, 0x8a, 0xa6, 0xfd, 0x29, 0x35, 0xd2,
-			0x41, 0x4c, 0xf0, 0xce, 0x4f, 0x74, 0x8a, 0x13, 0xbb, 0x28, 0x33, 0x21, 0x4c, 0x3b,
-			0x94, 0xa6, 0xd3, 0xb3,
+			0x0cu8, 0x67u8, 0xcau8, 0xf4u8, 0x61u8, 0x29u8, 0x0cu8, 0xd4u8, 0x63u8, 0xe5u8, 0x35u8,
+			0x21u8, 0x3fu8, 0x99u8, 0x6eu8, 0x32u8, 0x73u8, 0x6eu8, 0x65u8, 0xa2u8, 0x06u8, 0x37u8,
+			0x83u8, 0xfdu8, 0xe5u8, 0x03u8, 0x6bu8, 0x71u8, 0x39u8, 0x6du8, 0xfbu8, 0x0cu8,
 		];
 
-		// proof of alice is 0x5182a73e48bd6e814d0c2b41672d9cb8c87c4221b55bc08e0943198e90caad1f
+		// proof of alice is 0xd8b63c7168eef1bc3b00cdf73d1636429a26ab607b52f1de073b1f53edd9302d
 		let alice_proof = vec![[
-			0x51u8, 0x82u8, 0xa7u8, 0x3eu8, 0x48u8, 0xbdu8, 0x6eu8, 0x81u8, 0x4du8, 0x0cu8, 0x2bu8,
-			0x41u8, 0x67u8, 0x2du8, 0x9cu8, 0xb8u8, 0xc8u8, 0x7cu8, 0x42u8, 0x21u8, 0xb5u8, 0x5bu8,
-			0xc0u8, 0x8eu8, 0x09u8, 0x43u8, 0x19u8, 0x8eu8, 0x90u8, 0xcau8, 0xadu8, 0x1fu8,
+			0xd8u8, 0xb6u8, 0x3cu8, 0x71u8, 0x68u8, 0xeeu8, 0xf1u8, 0xbcu8, 0x3bu8, 0x00u8, 0xcdu8,
+			0xf7u8, 0x3du8, 0x16u8, 0x36u8, 0x42u8, 0x9au8, 0x26u8, 0xabu8, 0x60u8, 0x7bu8, 0x52u8,
+			0xf1u8, 0xdeu8, 0x07u8, 0x3bu8, 0x1fu8, 0x53u8, 0xedu8, 0xd9u8, 0x30u8, 0x2du8,
 		]];
 
-		run_to_block(1);
-		assert_eq!(System::block_number(), 1);
-
-		let _ = Balances::deposit_creating(&alice_account, (CREATION_FEE + 10).into());
+		// 0x84e8a1146740344fc59360423dcb97f58a6aa08e111b552c0a685af58c9bbb6d
+		let _bob_proof = vec![[
```

Comment:
Why we need this

Reply:
Just in case, people wonder what's merkleProof of bob and in case we reuse this in future.

Reply:
If we don't need it now, and you cannot make an issue to use this, no reason to leave it here

Reply:
_bob_proof  deleted